src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.StaticWebAssets.targets

```diff
@@ -0,0 +1,246 @@
+
+    <!-- Computes a default safe base path from the $(PackageId) that will be a prefix
+         to all the resources being exported from this library by default. The convention
+         consists of removing intermediate whitespaces, dots and lower casing all letters
+         in the package id using an invariant culture.
+
+         We don't aim to handle all possible cases for this prefix, as it can get really
+         complicated (non-unicode characters for example), so for that case,
+         StaticWebAssetBasePath can be set explicitly and we won't interfere.
+    -->
+    <GetDefaultStaticWebAssetsBasePath 
+      BasePath="$(PackageId)"
+      Condition="'$(StaticWebAssetBasePath)' == ''">
+      <Output TaskParameter="SafeBasePath" PropertyName="_StaticWebAssetSafeBasePath" />
+    </GetDefaultStaticWebAssetsBasePath>
+
+    <PropertyGroup>
+      <StaticWebAssetBasePath Condition="$(StaticWebAssetBasePath) == ''">_content/$(_StaticWebAssetSafeBasePath)</StaticWebAssetBasePath>
+    </PropertyGroup>
+
+    <ItemGroup>
+
+      <_ThisProjectStaticWebAsset 
+        Include="$(MSBuildProjectDirectory)\wwwroot\**"
+        Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
+      <!-- 
+        Should we promote 'wwwroot\**'' to a property?
+        We don't want to capture any content outside the content root, that's why we don't do
+        @(Content) here.
+      -->
+      <StaticWebAsset Include="@(_ThisProjectStaticWebAsset)">
+        <!-- (Package, Project, '' (CurrentProject)) -->
+        <SourceType></SourceType>
+        <!-- Identifier describing the source, the package id, the project name, empty for the current project.  -->
+        <SourceId>$(PackageId)</SourceId>
+        <!-- 
+          Full path to the content root for the item:
+          * For packages it corresponds to %userprofile%/.nuget/packages/<<PackageId>>/<<PackageVersion>>/razorContent
+          * For referenced projects it corresponds to <<FullProjectRefPath>>/wwwroot
+          * For the current projects it corresponds to $(MSBuildThisProjectFileDirectory)wwwroot\
+        -->
+        <ContentRoot>$(MSBuildProjectDirectory)\wwwroot\</ContentRoot>
+        <!-- Subsection (folder) from the url space where content for this library will be served. -->
+        <BasePath>$(StaticWebAssetBasePath)</BasePath>
+        <!-- Relative path from the content root for the file. At publish time, we combine the BasePath + Relative
+  path to determine the final path for the file. -->
+        <RelativePath>%(RecursiveDir)%(FileName)%(Extension)</RelativePath>
+
+      </StaticWebAsset>
+    </ItemGroup>
+
+    <!-- StaticWebAssets from referenced projects. -->
+    
+    <MSBuild
+      Condition="@(_StaticWebAssetsProjectReference) != ''"
+      Projects="@(_StaticWebAssetsProjectReference)"
+      BuildInParallel="$(BuildInParallel)"
+      ContinueOnError="!$(BuildingProject)"
+      Targets="GetCurrentProjectStaticWebAssets"
```

Comment:
Would it be nicer if the target was changed to return all static web assets and this target filtered and transformed it? Right now it seems like `GetCurrentProjectStaticWebAssets` target is expected to shape the input for this target.

```xml
<Target Name="GetStaticWebAssetItems" Outputs="@(StaticWebAsset)" />

...

<Output TaskParameter="TargetOutputs" ItemName="_ReferencedProjectStaticWebAssets" />

<ItemGroup>
   <StaticWebAsset Include="@(_ReferencedProjectStaticWebAssets)" Condition="%(_ReferencedProjectStaticWebAssets.SourceType) == ''" SourceType="Project" />
</ItemGroup>
```

Reply:
No, that would make the target do additional work, because we would include items transitively and duplicates, its simpler if each project just returns its own static assets and the top level project collects everything.